src/main/java/graphql/ExecutionInput.java

```diff
@@ -379,8 +393,21 @@ public Builder dataLoaderRegistry(DataLoaderRegistry dataLoaderRegistry) {
             return this;
         }
 
+        /**
+         * @param incrementalSupport whether the execution has support for incremental delivery of data, via the @defer and @stream directives.
+         * <p>
+         * This is currently an experimental feature, and only @defer is supported.
+         *
+         * @return this builder
+         */
+        @ExperimentalApi
+        public Builder incrementalSupport(boolean incrementalSupport) {
+            this.incrementalSupport = incrementalSupport;
+            return this;
+        }
+
```

Comment:
What is this boolean for on input? - is it a feature flag of sorts ?

if so I think we should use GraphqlContext- we do in other cases - we not quite that but we inject specific code like `graphql.execution.values.InputInterceptor` say

Reply:
> is it a feature flag of sorts ?

That's right. It allows Graphql servers to decide whether they want to enable @defer`/`@stream` for a certain query.

> I think we should use GraphqlContext
So the value would be store on the map under a special key? Client code would do something like
```
graphqlContext.put("incrementalSupport", true); 
graphqlContext.put(GraphQLContext.INCREMENTAL_SUPPORT, true); 
```

Reply:
right - rather than create a viral API (ExecutionIinput gets a new member, ExecutionContext gets one, Execition gets one....

we would do as uou have above from a well known constant and the code that checks if its on or not just looks for that magic name.

I like the FF idea - but adding more API to existing well known classes for a FF that may go away is painful.

GraphqlContext can be set more easily and its not viral in how it affects class chapes